### PR TITLE
Add an implementation of `Stop-Computer` for Linux and macOS

### LINKED ISSUE #5448: Missing Stop-Computer cmdlet
The Stop-Computer cmdlet is missing:

```
PS /home/leo> Stop-Computer                                                    
Stop-Computer : The term 'Stop-Computer' is not recognized as the name of a 
cmdlet, function, script file, or operable program. Check the spelling of the 
name, or if a path was included, verify that the path is correct and try 
again.
At line:1 char:1
+ Stop-Computer
+ ~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Stop-Computer:String) [], Comm 
   andNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

Would it be possible to either add it in, or document that it is not available?

### PR DESCRIPTION
**Pull request by brendandburns**
# PR Summary
Adds an implementation of the Stop-Computer cmdlet for Linux and OS X

## PR Context

Closes #5448

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
- **User-facing changes**
    - [X] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [X] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5171
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Comment by brendandburns** [comment]
I'm pretty sure the Windows CI run is a flake.

File: src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj]: Please don't prevent cross compilation.  Pass in the property from the build.psm1.  You can then default the property if it is not set from the commandline.

Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj]: Example:
https://github.com/PowerShell/PowerShell/blob/c17d5a7a2796cc31b504c65724b4f8a5292ee344/PowerShell.Common.props#L115
https://github.com/PowerShell/PowerShell/blob/c17d5a7a2796cc31b504c65724b4f8a5292ee344/build.psm1#L347

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comment

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Is there a reason start-process won't work?

Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: `-P` is not an option on the macOS shutdown

Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: How do we handle it when the user doesn't have permissions to shutdown?

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: I think you mean the C# Process class

Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Yes, here is an example: 
https://github.com/PowerShell/PowerShell/blob/9851b07b1def05f83ae89c78071956ef1b4dca3f/src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs#L19

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: It would be better to update the existing `Computer.cs` and `#if UNIX` sections that apply to non-Windows (removing the global `#if !UNIX` at the top).  This way `Restart-Computer` is also just a call to the `shutdown` command.

**Comment by TravisEz13** [comment]
@brendandburns I had the CI system rerun the flaky tests and filed: https://github.com/PowerShell/PowerShell/issues/11168

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: It looks like this field is never used.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: I don't think try/catch is necessary here.  Dispose() can be called multiple times (I believe `ObjectDisposedException` is only thrown when a disposed object is used).

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Please put error messages in resx file, so that they can be localized when we support that (ErrorRecord FullyQualifiedErrorId string is not localized and need not go in a resx file).

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Should the compare to "localhost" be case insensitive?  It seems like it should to be compatible with other PowerShell cmdlets.

**Review by PaulHigin — CHANGES_REQUESTED** [review]
Overall looks good.  I would like to see Restart-Computer also be enabled for non-Windows platforms.

File: src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj]: Fixed to be a runtime thing.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: I tried this. >90% of Computer.cs is Windows specific. So the code becomes really, really ugly. 

It's your call, ultimately. It's do-able, but ugly imho. Please let me know if you still want me to do it.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: removed.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: This is copy/paste from `Computer.cs` so if that's true it should be fixed in both places, I'm not enough of a C# expert to know which is right.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: done.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: done.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: done.

Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: fixed.

Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: The command will error out and we (now) write the error code to the stream.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: I could also splint into:

* `Computer.cs`
* `Computer.Windows.cs`
* `Computer.Unix.cs`

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: The documentation I found says Dispose() 'typically' does not throw the `ObjectDisposedException`.  I seem to remember other documentation saying Dispose() should not throw.  But most (I guess not all :)) of our code base assumes it does not throw and I think we should follow that pattern here.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Missed this path, where we can use the same ComputerResources.LocalhostOnly message string.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (PaulHigin) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Since StopProcessing() is asynchronous, we should probably catch ObjectDisposedException here.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: done.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: this is actually covered by `InvalidOperationException`

**Comment by brendandburns** [comment]
comments addressed, please take another look.

**Review by PaulHigin — APPROVED** [review]
LGTM

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: If the majority of the code is Windows specific it would make it harder to maintain trying to combine them, so I agree to keep it separate.  Thanks for looking into it.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: This parameter is not valid for UNIX and should be removed

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: This parameter is also not valid for UNIX and should be removed

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: This parameter is also not valid for UNIX and should be removed

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: No longer applicable once the `-ComputerName` param is removed

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: I fixed it here (and in `Computer.cs`), please take another look.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Removed.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Removed

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Removed

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Removed (and removed the error message I added too)

Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs]: Also added a TODO to consider stopping remote machines (via ssh)

**Comment by brendandburns** [comment]
@SteveL-MSFT comments addressed, please take another look.

Thanks!

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Comment by brendandburns** [comment]
@TravisEz13 any chance you can take another look here, I'd love to see this merged since I have a follow up PR to share once it is merged.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@brendandburns Sorry, I've been out of the office.

**Pull request merged**
